torch/_dynamo/eval_frame.py

```diff
@@ -113,7 +116,27 @@ def _maybe_set_eval_frame(callback: DynamoCallback):
         )
         return callback
     else:
-        return set_eval_frame(callback)
```

Comment:
I think I'd prefer only going into the complicated logic when `warmup_runs` is non-zero